app/src/main/java/dk/gls/kotlin_emdk/EMDKViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package dk.gls.kotlin_emdk
+
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dk.gls.kemdk.DeviceSerialUtil
+import dk.gls.kemdk.EMDKResult
+import dk.gls.kemdk.EMDKThrowable
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class EMDKViewModel() : ViewModel() {
+
+    private val _deviceSerialFlow = MutableStateFlow("init")
```

Comment:
Consider adding a UIState with Initial, SerialNumber and Error sub classes instead of storing it in a primitive.